quipucords/tests/api/source/test_serializer.py

```diff
@@ -108,7 +108,7 @@ def test_openshift_source_default_port(openshift_cred_id):
     assert serializer.validated_data["port"] == 6443
 
 
-@pytest.mark.xpass("hosts ('5.4.3.2.1') is clearly invalid in this case")
+@pytest.mark.xfail(reason="hosts ('5.4.3.2.1') is clearly invalid in this case")
```

Comment:
I don't understand. Shouldn't this test use [`pytest.raises`](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions) to check that correct exception is raised when incorrect data is provided?

xfail is mostly intended for TDD (where you have a test for something that is not yet implemented) and heavily-parameterized tests where certain combinations are invalid.

Reply:
no exception is raised in this test :/

Reply:
but I agree with you, if the test actually failed (as it should, given your point in the other comment), we should have no markers on this one and use pytest.raises on the statement that would raise the failure.

Reply:
Correct. This test _should have always_ failed, but it _doesn't_ fail.

The "right" thing to do here would be to fix the underlying application logic so that invalid IP addresses are rejected. If we do that, then this test would _actually_ fail, prompting us to rewrite the test.

This is all just a result of bad old code. I haven't touched any of this code. It's just there, and this `xpass` "marker" has been wrong ever since it was implemented. I'm just fixing the marker to do the right thing in these circumstances.

The net result is that amid the `......` of the test output, this test will now say `X` instead of `.`. It will not change the result code of the test. It still "passes" but it a) no longer emits this warning message about an invalid marker, and b) clearly indicates that the test doesn't behave as it _should_.